java/ql/src/utils/model-generator/CaptureSourceModels.ql

```diff
@@ -44,7 +44,8 @@ string captureSource(TargetAPI api) {
     config.hasFlow(source, sink) and
     sourceNode(source, kind) and
     api = source.getEnclosingCallable() and
```

Comment:
@bmuskalla : Do you know if it was deliberate to restrict source models to those where the existing source is directly in the callable that we generate the summary for? That will exclude wrapper methods, i.e.
```java
string source() { // will be included
  return knownSource();
}

string sourceWrapped() { // will not be included
  return source();
}
```

If not, I would expect that `api = source.getEnclosingCallable()` should be removed.

Reply:
No, that was not intentional. But also not covered [by the tests](https://github.com/github/codeql/blob/main/java/ql/test/utils/model-generator/p/Sources.java) so it might be good to have a case of that

Reply:
@michaelnebel : Could you add a test that verifies my suspicion? And if I am correct, then remove the `api = source.getEnclosingCallable()` line?

Reply:
@hvitved: That sounds like an improvement we can try and make, when the implementation is ported.
Should we just merge this PR now?

Reply:
OK, as long as we also backport the improvement to Java.

Reply:
Looks like we wrote stuff at the same time :-) I will try and make the improvement.